src/components/SelectBadge/SelectBadge.tsx

```diff
@@ -84,41 +72,48 @@ export interface Props extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onCha
 
 const { block, elem } = bem('SelectBadge', styles);
 
-export const SelectBadge: React.FC<Props> = ({
+export function SelectBadge<V, O>({
     children,
     isDisabled = false,
-    onChange,
+    onChange = undefined,
     onDelete = undefined,
     onPriorityChange = undefined,
     option,
+    optionToLabel,
+    optionToKey,
     optionList,
     optionListHeader,
-    priority = 'mandatory',
-    priorityLabels,
+    priorityItem,
+    priorityItems,
     priorityButtonLabel,
     optionButtonLabel,
     deleteButtonLabel,
     ...rest
-}) => {
+}: Props<V, O>) {
```

Comment:
Let's think about props how we going to use it, and make them mandatory if we need to use some features, for example if exist `option`, then `optionToKey` is mandatory.
```
// Grouping the priority-related props
type PriorityProps<V> {
    items: Array<PriorityItem<V>>;
    selectedItem: PriorityItem<V>;
    buttonLabel: string;
    onChange: (newPriorityItem: PriorityItem<V>) => void;
}

// Grouping the option-related props
type OptionProps<O> {
    selectedOption: O;
    toLabel: (option: O) => string;
    toKey: (option: O) => string;
    list: Array<O>;
    listHeader?: string;
    buttonLabel: string;
    onChange: (newOption: O) => void;
}

// The base props common to all cases
export interface BaseProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'> {
    children: React.ReactNode;
    onDelete?: (e: React.KeyboardEvent | React.MouseEvent) => void;
    isDisabled?: boolean;
    deleteButtonLabel: string;
    priority?: PriorityProps<V>;
    option?: OptionProps<O>;
}
```
What do you think? But I don't know how it will be work with Storybook docs

Reply:
I love the idea, it was getting too confusing...  Storybook was not very happy, but I found way around it 🤞